src/tests/conformance/messages.rs

```diff
@@ -349,6 +346,108 @@ async fn basic_query_response_unseeded() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn disconnects_for_trivial_issues() {
+    // ZG-CONFORMANCE-011
+    //
+    // The node disconnects for trivial (non-fuzz, non-malicious) cases.
+    //
+    // - `Ping` timeout.
+    // - `Pong` with wrong nonce.
+    // - `GetData` with mixed types in inventory list.
+    // - `Inv` with mixed types in inventory list.
+    // - `Addr` with `NetworkAddr` with no timestamp.
+    //
+    // Note that the Addr test requires commenting out the relevant code in the encode
+    // function of NetworkAddr as we cannot encode without a timestamp.
+    //
+    // This test currently fails for zcashd and zebra.
+    //
+    // Current behaviour:
+    //
+    //  zcashd:
+    //      Ping(timeout)   - does not timeout after 2 minutes
+    //      GetData(mixed)  - responds to both
+    //      Inv(mixed)      - ignores the message
+    //      Addr            - Reject(Malformed), but no DC
+    //
+    //  zebra:
+    //      Pong            - ignores the message
+
+    // Create a node and main connection
+    let mut node: Node = Default::default();
+    node.initial_action(Action::WaitForConnection(new_local_addr()))
+        .log_to_stdout(true)
+        .start()
+        .await;
+
+    // Ping with timeout
+    let filter = MessageFilter::with_all_auto_reply().with_ping_filter(Filter::Disabled);
+    let mut stream = initiate_handshake(node.addr()).await.unwrap();
+    match filter.read_from_stream(&mut stream).await.unwrap() {
+        Message::Ping(_) => {
+            match timeout(
+                Duration::from_secs(120),
```

Comment:
The timeout for `zcashd` is set here https://github.com/zcash/zcash/blob/49ffee3f20b972dc3aa75e422c67523251cf088b/src/net.h#L47. We should have a test case which waits 20 mins, let's remove it for now and comment why.